controllers/replication/secret_controller.go

```diff
@@ -224,15 +225,21 @@ func (s *SecretSyncController) Reconcile(ctx context.Context, req ctrl.Request)
 			} else {
 				namespace = target.Namespace
 			}
+			if remoteClient == localClient && target.Namespace == sec.Namespace {
+				//TODO: @Michael does this allow us to remove the UID check in requiresUpdate or any other logic?
```

Comment:
No, this type of check shouldn't be here. If there's a prefix feature, then the copy could target the same namespace also in the same cluster.

Also, in some cases localClient could be different from remoteClient even if they target the same cluster. At least if the localClient is set to be non-caching client which the remoteClient isn't.

Lets say we have "my-user-supplied-A" and we want to copy it always for reason X to "cluster1-my-user-supplied-A". With this check in place, that won't be possible.

Reply:
I'm removing this check as advised.